Neos.Neos/Classes/Controller/Backend/ContentController.php

```diff
@@ -129,37 +121,33 @@ public function initializeUploadAssetAction()
      * Depending on the $metadata argument it will return asset metadata for the AssetEditor
      * or image metadata for the ImageEditor
      *
+     * Note: This action triggers the AssetUploaded signal that can be used to adjust the asset based on the
+     * (site) node it was attached to.
+     *
      * @param Asset $asset
      * @param string $metadata Type of metadata to return ("Asset" or "Image")
+     * @param NodeInterface $node The node the new asset should be assigned to
+     * @param string $propertyName The node property name the new asset should be assigned to
+     * @param string $siteNodeName The node name of the site the asset belongs to
      * @return string
      */
-    public function uploadAssetAction(Asset $asset, $metadata)
+    public function uploadAssetAction(Asset $asset, string $metadata, NodeInterface $node, string $propertyName, string $siteNodeName)
```

Comment:
hmm just thought why we send the $siteNodeName here? that can be extrapolated from the node.. this would make it more generic and less focussed on assigning the node to the site collection..

would also make it easier to implement the signal for people..

Reply:
@aertmann true. I thought about it as well and didn't want to change too much - but now that it got a bigger change anyways..
`$node->getContext()->getCurrentSite()` would be the way to go I assume?!
I'll push a follow-up, hopefully the last one ;)